kernel/enter.c

```diff
@@ -95,6 +95,82 @@ static int _process_mount_configuration(myst_mounts_config_t* mounts)
     return ret;
 }
 
```

Comment:
Does the `/etc` directory created need to have execution permission? I prefer least privilege approach.

If '/etc' directory already exist in the CPIO ramfs or the dm_verify ext2fs, is it possible that the code fails to create the `reslov.conf` file in that directory due to permission limitation?

In debug mode, Mystikos supports app rootfs on the hostfs (for testing/debugging purpose). Is the implementation compatible with that configuration?

Reply:
- The execution bit of a directory means one can access (read, write, execute) items within the directory. Without the execution permission of `/etc`, one cannot read `resolv.conf` under.
- If `/etc` directory doesn't have write permission, then creat() will fail. Currently, myst "don't check permissions fully yet", so even if the `/etc` directory doesn't have write permission, we can still creat resolv file under it. Also, in case rootfs already contains `/etc/resolv.conf` and it's readonly, we first unlink it (see https://github.com/deislabs/mystikos/pull/449/files#diff-e71a12ac51b1964e823bd4db8de920dcff6f742035713f1a76a53457977a91daR122). This is because "If you have write + execute permissions on a directory, you can {delete,rename} items living within even if you don't have write perimission on those items. (use sticky bit to prevent this)" https://unix.stackexchange.com/questions/21251/execute-vs-read-bit-how-do-directory-permissions-in-linux-work
- Good catch! The unlink() call failed. This was not exposed in ltp test because its appdir/ doesn't have appdir/etc/resolv.conf
```
$ myst exec-sgx  appdir /curl https://www.microsoft.com
kernel: failed to unlink file /etc/resolv.conf
```

Reply:
When we use hostfs as rootfs, we usually point the root directory to a folder on the host (which should be different from `/`), I think it's OK as long as we give a general warning about the danger of persisted file modification for users of hostfs.